lib/cisco_node_utils/bgp.rb

```diff
@@ -327,10 +327,19 @@ def enforce_first_as
       config_get('bgp', 'enforce_first_as', @get_args)
     end
 
-    def enforce_first_as=(state)
-      @set_args[:state] = (state ? '' : 'no')
-      config_set('bgp', 'enforce_first_as', @set_args)
-      set_args_keys_default
+    def enforce_first_as=(enable)
+      if platform == :ios_xr
+        # XR enforce_first_as is on by default (=>true)
+        # =>true  = enable enforce_first_as  = 'no bgp enforce-first-as disable'
+        # =>false = disable enforce_first_as = 'bgp enforce-first-as disable'
+        @set_args[:state] = (enable ? 'no' : '')
+        config_set('bgp', 'enforce_first_as', @set_args)
+        set_args_keys_default
+      else
+        @set_args[:state] = (enable ? '' : 'no')
+        config_set('bgp', 'enforce_first_as', @set_args)
+        set_args_keys_default
```

Comment:
Same request for the config_set call too - just the set_args should be different.